src/main/java/calculator/model/Delimiter.java

```diff
@@ -0,0 +1,62 @@
+package calculator.model;
+
+import calculator.constant.ErrorMessage;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+public class Delimiter {
+    private static final int STRICT_LENGTH = 1;
+    private static final String EMPTY = "";
+    private static final String SELECT_ONE_FORMAT = "[%s]";
+    private static final Pattern SINGLE_NUMERIC_PATTERN = Pattern.compile("^[0-9]$");
+
+    private final String character;
+
+    private Delimiter(String character) {
+        validateLength(character);
+        validateNotNumber(character);
+
+        this.character = character;
+    }
+
+    public static Delimiter from(String character) {
+        return new Delimiter(character.trim());
+    }
+
+    private void validateLength(String character) {
+        if (character.isBlank() || character.length() > STRICT_LENGTH) {
+            throw new IllegalArgumentException(ErrorMessage.DELIMITER_MUST_LENGTH_1.getMessage());
+        }
+    }
+
+    private void validateNotNumber(String character) {
+        if (SINGLE_NUMERIC_PATTERN.matcher(character).matches()) {
+            throw new IllegalArgumentException(ErrorMessage.DELIMITER_MUST_CHARACTER.getMessage());
+        }
+    }
+
+    public static String toRegex(List<Delimiter> delimiters) {
+        return String.format(
+                SELECT_ONE_FORMAT,
+                String.join(EMPTY, delimiters.stream().map(delimiter -> delimiter.character).toList())
+        );
+    }
```

Comment:
```suggestion
public static String toRegex(List<Delimiter> delimiters) {
    return String.format(
        SELECT_ONE_FORMAT,
        String.join(
            EMPTY,
            delimiters.stream()
                      .map(delimiter -> delimiter.character)
                      .toList()
        )
    );
}

```

메서드 체인을 줄바꿈으로 구분하면 더 가독성이 좋을 것 같습니다! 스트림 연산을 따로 메서드로 빼는 것도 좋아보이네요!

Reply:
인덴트가 깊어지는 것 같아 한 줄에 썼는데, 역시나 가독성이 안 좋아 보이네요 🥲
가독성을 우선으로 생각하고, 인덴트가 깊어진다면 메서드 분리를 해야겠어요